File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        currentSum = 0
        result = nums[0]
        
        for num in nums:
            if currentSum < 0:
                currentSum = 0
            currentSum += num
            result = max(result, currentSum)
        
        return result

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int currentSum = 0;
        int result = nums[0];
        
        for(int num : nums) {
            
            if (currentSum < 0) {
                currentSum = 0;
            }
            
            currentSum += num;
            
            result = max(result, currentSum);
        }
        
        return result;
        
    }
};